be/src/formats/parquet/file_reader.cpp

```diff
@@ -67,36 +67,58 @@ Status FileReader::init(HdfsScannerContext* ctx) {
 }
 
 Status FileReader::_parse_footer() {
-    std::vector<char> footer_buffer;
-    ASSIGN_OR_RETURN(uint32_t footer_read_size, _get_footer_read_size());
-    footer_buffer.resize(footer_read_size);
-
-    {
-        SCOPED_RAW_TIMER(&_scanner_ctx->stats->footer_read_ns);
-        RETURN_IF_ERROR(_file->read_at_fully(_file_size - footer_read_size, footer_buffer.data(), footer_read_size));
-        _scanner_ctx->stats->request_bytes_read += footer_read_size;
-        _scanner_ctx->stats->request_bytes_read_uncompressed += footer_read_size;
-    }
-
-    ASSIGN_OR_RETURN(uint32_t metadata_length, _parse_metadata_length(footer_buffer));
+    std::string footer_buffer;
+    uint32_t footer_size = 0;
```

Comment:
I think `metadata_length` is better. Because the parquet footer contains metadata + 8bytes, `footer_size` is only used to deserialize metadata thrift.